src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -371,6 +390,16 @@ internal override ImmutableArray<Cci.INamespaceTypeDefinition> GetAnonymousTypes
             return StaticCast<Cci.INamespaceTypeDefinition>.From(Compilation.AnonymousTypeManager.GetAllCreatedTemplates());
         }
 
+        internal override ImmutableArray<Cci.INamespaceTypeDefinition> GetMethodGroupConversionCacheFrames()
+        {
+            if (EmitOptions.EmitMetadataOnly)
+            {
+                return ImmutableArray<Cci.INamespaceTypeDefinition>.Empty;
+            }
+
+            return StaticCast<Cci.INamespaceTypeDefinition>.From(Compilation.MethodGroupConversionCacheFrameManager.GetAllCreatedFrames());
```

Comment:
Type casts cannot be used to convert symbols to CCI interfaces, symbols should be translated. Our symbol types claim to implement many different CCI interfaces for efficiency, but it doesn't mean the implementation is correct for all situations, therefore, there is a translation layer that decides if it is appropriate for a certain symbol to implement some interface directly or requires a wrapper, etc.